src/lib/libexceptions.js

```diff
@@ -277,17 +278,23 @@ var LibraryExceptions = {
     return [type, message];
   },
 #endif
+
 #if WASM_EXCEPTIONS
+#if ASSERTIONS
+  // Try to give a more useful error message than simply `Undefined reference to `emscripten_longjmp`
+  emscripten_longjmp__deps: [() => error('undefined reference to `emscripten_longjmp`. One or more object files was not compiled with `-fwasm-exceptions`.  Build with `-sASSERTIONS=0` to have wasm-ld report which one.')],
```

Comment:
Just to confirm, `-sASSERTIONS=0` is the default right?

Reply:
`-sASSERTIONS=1` is the default for `-O0` but `-sASSERTIONS=0` is the default of `-O1` and above.